server/src/main/java/org/elasticsearch/common/logging/ClusterIdConverter.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.common.logging;
+
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.core.config.plugins.Plugin;
+import org.apache.logging.log4j.core.pattern.ConverterKeys;
+import org.apache.logging.log4j.core.pattern.LogEventPatternConverter;
+import org.apache.logging.log4j.core.pattern.PatternConverter;
+
+/**
+ * Pattern converter to format the cluster_id variable into JSON fields <code>cluster.id</code>.
+ */
+@Plugin(category = PatternConverter.CATEGORY, name = "ClusterIdConverter")
+@ConverterKeys({"cluster_id"})
+public final class ClusterIdConverter extends LogEventPatternConverter {
```

Comment:
I wonder if this would be more clear if we put a comment saying that this is used for `ECSJsonLayout`
it is probably not usual to comment about the usage of the class in its javadoc, but it might not be clear at first why there are `ClusterIdConverter` and `NodeAndClusterIdConverter` at the same time.
WDYT?

Reply:
Yes, I think that's a good idea. I've added a comment in both convertor classes.